src/dotnet/Chat.UI.Blazor/Components/PlaceSettings/PlaceSettingsEditTypeModalPage.razor

```diff
@@ -0,0 +1,224 @@
+
+    private Session Session => Hub.Session();
+    private IPlaces Places => Hub.Places;
+
+    [CascadingParameter] public DiveInModalPageContext Context { get; set; } = null!;
+
+    private PlaceId PlaceId { get; set; }
+
+    protected override void OnInitialized() {
+        PlaceId = Context.GetTypedModel<PlaceId>();
+        Context.SetTitle("Edit place settings");
+        _submitButtonInfo = DialogButtonInfo.CreateSubmitButton("Save", OnSubmit);
+        Context.RegisterButtons(DialogButtonInfo.CancelButton, _submitButtonInfo);
+        base.OnInitialized();
+    }
+
+    protected override async Task OnInitializedAsync() {
+        var place = await Places.Get(Session, PlaceId, default).Require();
+        _form = new FormModel(ComponentIdGenerator) {
+            IsPublic = place.IsPublic,
+        };
+        Context.SetClass("edit-chat-type");
+        await base.OnInitializedAsync();
+    }
+
+    protected override ComputedState<ComputedModel>.Options GetStateOptions()
+        => ComputedStateComponent.GetStateOptions(GetType(),
+            static t => new ComputedState<ComputedModel>.Options() {
+                InitialValue = ComputedModel.Loading,
+                Category = ComputedStateComponent.GetStateCategory(t),
+            });
+
+    protected override async Task<ComputedModel> ComputeState(CancellationToken cancellationToken) {
+        var placeId = PlaceId;
+
+        var place = await Places.Get(Session, placeId, default).Require();
+        if (!place.Rules.CanEditProperties())
+            return new() { Place = place };
+
+        var invites = await Hub.Invites.ListPlaceInvites(Session, placeId, cancellationToken);
+        var threshold = Hub.Clocks().SystemClock.Now - TimeSpan.FromDays(3);
+        var activeInvites = invites
+            .Where(c => c.ExpiresOn > threshold)
+            .OrderByDescending(c => c.ExpiresOn)
+            .ToList();
+        var welcomeChatId = await Places.GetWelcomeChatId(Session, placeId, cancellationToken).ConfigureAwait(false);
+        var link = !welcomeChatId.IsNone
+            ? Links.Chat(welcomeChatId).ToDisplayUrl(Hub.UrlMapper())
+            : (DisplayUrl?)null;
+        var isPublic = place.IsPublic;
+        return new() {
+            Place = place,
+            Link = link,
+            Invites = activeInvites,
+            IsPublic = isPublic,
+        };
+    }
+
+    private async Task OnNewInviteClick() {
+        var invite = Invite.New(Constants.Invites.Defaults.PlaceRemaining, new PlaceInviteOption(PlaceId));
+        invite = await Hub.UICommander().Run(new Invites_Generate(Session, invite));
+        _newInviteId = invite.Id;
+    }
+
+    private async Task OnSubmit() {
+        if (_formRef == null)
+            return;
+        if (!_formRef.IsValid)
+            return;
+        await Save();
+        Context.Close();
+    }
+
+    private async Task Save() {
+        var place = await Hub.Places.Get(Session, PlaceId, default).Require();
```

Comment:
Use Places shorcut instead of Hub.Places

Reply:
Done